shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/check/datasource/AbstractDataSourceCheckerTest.java

```diff
@@ -64,19 +64,41 @@ public void checkVariable(final Collection<? extends DataSource> dataSources) {
             }
         };
         dataSources = new LinkedList<>();
+        RuleAlteredJobContext jobContext = new RuleAlteredJobContext(ResourceUtil.mockJobConfig());
+        dataSource = new PipelineDataSourceManager().getDataSource(jobContext.getTaskConfig().getDumperConfig().getDataSourceConfig());
         dataSources.add(dataSource);
     }
     
     @Test
-    public void assertCheckConnection() throws SQLException {
-        when(dataSource.getConnection()).thenReturn(connection);
+    public void assertCheckConnection() {
         dataSourceChecker.checkConnection(dataSources);
-        verify(dataSource).getConnection();
     }
     
     @Test(expected = PipelineJobPrepareFailedException.class)
     public void assertCheckConnectionFailed() throws SQLException {
-        when(dataSource.getConnection()).thenThrow(new SQLException("error"));
-        dataSourceChecker.checkConnection(dataSources);
+        DataSource mockDatasource = mock(DataSource.class);
+        when(mockDatasource.getConnection()).thenThrow(new SQLException("error"));
+        dataSourceChecker.checkConnection(Collections.singletonList(mockDatasource));
+    }
+    
+    @Test
+    public void assertCheckTargetTable() throws SQLException {
+        try (Connection connection = dataSource.getConnection();
+             Statement statement = connection.createStatement()) {
+            statement.execute("DROP TABLE IF EXISTS t_order");
+            statement.execute("CREATE TABLE t_order (order_id INT PRIMARY KEY, user_id VARCHAR(12))");
+        }
+        dataSourceChecker.checkTargetTable(dataSources, Collections.singletonList("t_order"));
+    }
+    
+    @Test(expected = PipelineJobPrepareFailedException.class)
+    public void assertCheckTargetTableFailed() throws SQLException {
+        try (Connection connection = dataSource.getConnection();
+             Statement statement = connection.createStatement()) {
+            statement.execute("DROP TABLE IF EXISTS t_order");
+            statement.execute("CREATE TABLE t_order (order_id INT PRIMARY KEY, user_id VARCHAR(12))");
+            statement.execute("INSERT INTO t_order (order_id, user_id) VALUES (1, 'xxx'), (999, 'yyy')");
+        }
+        dataSourceChecker.checkTargetTable(dataSources, Collections.singletonList("t_order"));
     }
```

Comment:
Could we test it with mocked `dataSource`, since `assertCheckConnectionFailed` already use mocked dataSource.